app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt

```diff
@@ -0,0 +1,30 @@
+package compiler.intermediate_form
+
+import compiler.ast.Variable
+
+data class FunctionDetailsGenerator(
+    val depth: Int,
+    val vars: Map<Variable, Boolean>,
+    val parameters: List<Variable>
+)
+
+fun generateCall(args: List<IntermediateFormTreeNode>): ControlFlowGraph {
```

Comment:
Shouldn't we also return an IFTNode for function result, outside that CFG: `Pair<ControlFlowGraph, IntermediateFormTreeNode?>`?